cloudmarker/checks/firewallrulecheck.py

```diff
@@ -0,0 +1,119 @@
+"""Firewall Rule Check plugin for GCP Firewall rule record.
+
+This modules define a :class:`FirewallRuleCheck` that identifies weak Firewall
+rules for Google Cloud Platform.
+"""
+
+
+class FirewallRuleCheck:
+    """Firewall Rule Check plugin.
+
+    This class identifies weak Firewall rules in the record objects by
+    comparing the them with predefined weak Firewall rules
+    configurations.
+    """
+
+    def __init__(self):
+        """Create instance of :class:`FirewallRuleCheck` plugin."""
+        # self._misconfigurations defines the set of weak/misconfigured
+        # firewall rules. Given any firewall rule, is compared with
+        # these predefined weak misconfigurations if the rule matches to
+        # any one predefined rule, it will be marked as weak and will be
+        # reported as anomaly. Adding IPProtocol field have been skipped
+        # to check for misconfiguration as GCP docs says that: The
+        # protocol type is required when creating a firewall rule. This
+        # value can either be one of the following well known protocol
+        # strings (tcp, udp, icmp, esp, ah, ipip, sctp), or the IP
+        # protocol number, implying it cannot be *.
+        self._misconfigurations = {
+            'sourceRanges': [
+                '0.0.0.0/0',
+                '*'
+            ],
+            'destinationRanges': [
+                '0.0.0.0/0',
+                '*'
+            ],
+        }
+
+    def _process_rule(self, record):
+        _misconfigurations = []
+
+        for misconfig_field in self._misconfigurations:
+            if misconfig_field in record:
+                for record_misconfig_field in record[misconfig_field]:
+                    if (record_misconfig_field in
+                            self._misconfigurations[misconfig_field]):
+                        _misconfigurations.append(misconfig_field
+                                                  + ':'
+                                                  + record_misconfig_field)
```

Comment:
Please rename some of the variables in the above code so that it is easier to look at the variable and tell what it represents (a key or a value? within the rules dictionary or within the actual firewall record dictionary?). Here are my suggestions:

- `self._misconfigurations` to `self._rules`
- `misconfig_field` to `rule_key`
- `record_misconfig_field` to `record_item`